apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2835,3 +2860,99 @@ func (u *UniterAPI) APIAddresses(ctx context.Context) (result params.StringsResu
 
 	return u.APIAddresser.APIAddresses(ctx, controllerConfig)
 }
+
+// WatchApplication starts an NotifyWatcher for an application.
+func (u *UniterAPIv20) WatchApplication(ctx context.Context, args params.Entities) (params.NotifyWatchResults, error) {
```

Comment:
question: Do we really want to make this a bulk call? For Watch we must to for compatibility. However we know a uniter worker is only handling a single unit, so only one application. 

`WatchApplication(ctx context.Context, arg param.Entity) (params.NotifyWatchResult, error)`

Reply:
Changed to be a single entity facade.